pkg/apis/cassandra.rook.io/v1alpha1/types.go

```diff
@@ -120,4 +120,29 @@ type RackStatus struct {
 	Members int32 `json:"members"`
 	// ReadyMembers is the number of ready members in the specific Rack
 	ReadyMembers int32 `json:"readyMembers"`
+	// Conditions are the latest available observations of a rack's state.
+	Conditions []RackCondition `json:"conditions,omitempty"`
 }
+
+// RackCondition is an observation about the state of a rack.
+type RackCondition struct {
+	Type   RackConditionType `json:"type"`
+	Status ConditionStatus   `json:"status"`
+}
+
+type RackConditionType string
+
+const (
+	RackConditionTypeMemberLeaving RackConditionType = "MemberLeaving"
+)
+
+type ConditionStatus string
```

Comment:
this is from https://github.com/kubernetes/api/blob/master/core/v1/types.go#L2229 right? any reason to duplicate it here instead of just referencing it? same for the `ConditionStatus` consts below

Reply:
I duplicated it in order to not have the dependency. 
The code is really small so I didn't see the reason to add an extra dependency there.

Reply:
@jbw976 On second thought, I think you are right on this one!
I changed the ConditionStatus type and consts to the ones from `k8s.io/api/core/v1`.